server.py

```diff
@@ -66,13 +69,49 @@ def index():
 @app.route('/inventory', methods=['GET'])
 def list_inventory():
     """ Return all entries in the Inventory system """
-    return jsonify(message='List all inventory API'), HTTP_200_OK
+    results = [entry.serialize() for entry in Entry.list_all()]
+    return jsonify(results), HTTP_200_OK
 
-@app.route('/inventory/<int:entry_id>', methods=['GET'])
-def get_entry(entry_id):
-    """ Return entry identified by entry_id """
-    message = {'message' : 'GET request for entry %s' % str(entry_id)}
-    return jsonify(message), HTTP_200_OK
+@app.route('/inventory/<int:prod_id>', methods=['GET'])
+def get_entry(prod_id):
+    """ Return entry identified by prod_id """
+    entry = Entry.find(prod_id)
+    if not entry:
+        raise NotFound("Product with id '{}' was not found in Inventory".format(prod_id))
+    return jsonify(entry.serialize()), HTTP_200_OK
+
+@app.route('/inventory', methods=['POST'])
+def create_entry():
+    """
+    Creates an Inventory entry
+    This endpoint will create an entry based the data in the body that is posted
+    """
+    check_content_type('application/json')
+    entry = Entry()
+    entry.deserialize(request.get_json())
+    entry.save()
+    message = entry.serialize()
+    location_url = url_for('get_entry', prod_id=entry.prod_id, _external=True)
+    return make_response(jsonify(message), HTTP_201_CREATED,
+                         {
+                             'Location': location_url
```

Comment:
what does "Location" here mean?

Reply:
The "Location" is a url included in the HTTP Request header pointing to the newly-created resource.